Sources/Signal.swift

```diff
@@ -24,8 +24,20 @@ public final class Signal<Value, Error: Swift.Error> {
 	/// when the signal terminates.
 	private var generatorDisposable: Disposable?
 
-	/// The state of the signal. `nil` if the signal has terminated.
-	private let state: Atomic<SignalState<Value, Error>?>
+	/// The status of the signal.
+	///
+ 	/// `status` synchronizes using Read-Copy-Update. Reads are thus wait-free,
+ 	/// but writes are required not to mutate in place. This suits `Signal` as
+ 	/// reads to `status` happens on the critical path of event delivery, while
+ 	/// adding and removing observers or termination generally has a constant
+	/// occurrence.
```

Comment:
You can disassemble using LLDB to verify this. For example on ARMv8, it looks like:
```
->  0x103005b24 <+528>:  ldr    x25, [x20, #32]
    0x103005b28 <+532>:  lsr    x8, x25, #62
    0x103005b2c <+536>:  cbnz   x8, 0x103005be8           ; <+724> at Signal.swift:169
```
with `x20` being the reference to `signal`, which is an aligned pointer, and the offset being 32 bytes. So the resulting address is still an aligned location.
```
     x20 = 0x00000001712649c0
```